acme/acme/client_test.py

```diff
@@ -138,7 +138,7 @@ def test_init_acme_version(self):
     def test_query_registration_client_v2(self):
         self.response.json.return_value = DIRECTORY_V2.to_json()
         client = self._init()
-        self.response.json.return_value = self.regr.body.to_json()
+        self.response.json.return_value = self.regr.body.to_json()  # pylint: disable=no-member
```

Comment:
I don't like having to disable checks like this. It's not a huge deal because there aren't many instances of this in this PR, but `pylint` is wrong and making noise because it can't properly understand our code. `pylint` is giving devs false information, causing tests to fail, and making us spend time reassuring it because it's not able to properly detect what's going on here.

Are there any better options? Is there any way to do something like add comments to objects in `messages.py` explaining to `pylint` the type/attributes of these objects?

Reply:
Alternatively, we can set the option `ignored-classes` in `.pylintrc`

Reply:
What other checks on these objects does that disable?

Reply:
Documentation of pylint states:


```
ignored-classes:
--
  | List of class names for which member attributes should not be checked (useful for classes with dynamically set attributes). This supports the use of qualified names. Default: optparse.Values,thread._local,_thread._local
```

https://docs.pylint.org/en/1.6.0/features.html#id5

So it seems to be really crafted for the situation of objects with dynamic parameters, and does not seems to removed any other check.

Reply:
However, regarding the low number of inline disables that are done to handle this issue, I think it is not necessary to remove every parameter check on these classes. It is still relevant to raise an error and be sure that the parameter effectively exists, before adding the inline comment to disable the check.

Reply:
Setting `ignored-classes` seems perfect to me.

I personally think it's worth looking through the `disable=no-member` lines (or [other messages from that checker](https://docs.pylint.org/en/1.6.0/features.html#id6) if it also affects that) and if `pylint` is wrong, adding it to the list of `ignored-classes`. I think setting it for the `Field` class in `josepy` and `contextlib.closing` gets most of them.

While this allows us to remove a few more disables from this PR, I'm mainly interested in doing this for future development. There's certainly an argument to be made for having `pylint` complain to you about things it's not sure about so the developer can double check, but I personally don't like this approach. I think it's another hurdle for new devs to try and understand and wastes all developers time. I'd rather have `pylint` check the things it can well and we'll rely on things like unit and integration tests to help and sure that these dynamic properties exist when `pylint` can't see them.

Reply:
@bmw I configured `ignore-classes` to remove JSON-related classes from the check, and remove most of the `no-member` suppress.